### PR TITLE
fix(observability-dashboards): calendar-align scoreboard budget-burn queries

### PR DESCRIPTION
**Pull request by stephane-segning**
## 1. Summary

This PR changes:

- `tools/dashboards/src/dashboards/envoy_ai_gateway/scoreboard.py`: the budget-burn gauge and the two spend stats (`_gauge_budget`, `_stat_spend_range` renamed from `_stat_spend_30d`, `_stat_budget_remaining`) now query `[$__range]` instead of a hardcoded `[30d]`.
- Dashboard default time range changed from `now-30d .. now` to `now/M .. now` ("this month so far").
- Regenerated `charts/observability-dashboards/files/envoy-ai-gateway/scoreboard.json` via `uv run dashboards build`.

It solves:

- The scoreboard's "Monthly budget burn" gauge and spend stats used a fixed 30-day rolling window, the same conceptual bug ADR-0111 fixed on the rate-limit enforcement side in #859 — a 30-day window is not a calendar month and drifts ~5 days/year off the real month boundary. Found while investigating ADR-0111.

---

## 2. Intent

> Make "monthly budget burn" on the App Scoreboard actually reset on the 1st of the calendar month instead of showing a rolling blend of the trailing 30 days (part last month, part this month). PromQL has no native "since start of month" range-vector duration, so the fix relies on the Grafana time picker's own relative-time resolution (`now/M`) driving `$__range`, matching the pattern every other panel in this file (and every other dashboard in `envoy_ai_gateway/`) already uses.

---

### In Scope

- The three `[30d]`-hardcoded PromQL queries in `scoreboard.py`.
- The dashboard's default time range.
- Updated docstring/comments that documented the old fixed-30d behavior as intentional.

### Out of Scope

- The Redis-derived `gateway_ratelimit_spend_micro_usd` quota dashboard (`ratelimit_quota.py`) — separate, already-filed follow-up (cross-repo, `ai-helm-values` `prometheus-redis-exporter` regex + a `$billing_period` variable). Unrelated data path (Redis live quota vs. Mimir historical cost).
- ADR-0111's gateway rate-limit enforcement fix — already merged, see #859.

---

## 4. Verification

I verified this change by:

- [x] Running automated tests
- [x] Checking logs
- [ ] Running manual tests
- [ ] Checking metrics
- [ ] Testing error cases
- [ ] Testing permissions/security behavior
- [ ] Testing rollback or failure behavior, if relevant

Commands run:

```bash
cd tools/dashboards
uv run dashboards build
uv run dashboards check
uv run ruff format . && uv run ruff check .
```

Results:

```text
wrote charts/observability-dashboards/files/envoy-ai-gateway/scoreboard.json
(+ 15 other unchanged dashboards re-serialized identically)
ok — every dashboard matches its generator
21 files left unchanged
All checks passed!
```

`git diff --stat` after build confirms only `scoreboard.json` changed.
Inspected the regenerated JSON directly: `time` is now `{"from": "now/M", "to": "now"}`
and all three panels' `targets[].expr` now end in `[$__range]` instead of `[30d]`.

Live confirmation that the gauge/stats actually read as calendar
month-to-date in the running Grafana (e.g. value drops near-zero right
after a month boundary) is **pending this PR's merge and ArgoCD sync**
— this repo's dashboards deploy via the `observability-dashboards`
chart reconciled by ArgoCD (continuous delivery, ADR-0055), so there is
no live dashboard to check against before merge. Will verify against
the live Grafana scoreboard after sync and report back.

---

## 5. Screenshots / Evidence

* Logs: build/check/lint output above.

---

## 6. Risk Assessment

Risk level:

* [x] Low

Potential risks:

* None to enforcement or data — this only changes a Grafana dashboard JSON (read-only visualization), no chart templates, no rate-limit config, no auth.
* Panel titles changed ("Spend (last 30d)" → "Spend (range)", "Monthly budget burn (last 30d)" → "Monthly budget burn (range)") — cosmetic only.

Mitigation:

* n/a — dashboard-only change, revertible by `git revert` (dashboard JSON is regenerated from source, no stateful migration).

---

## 7. AI Usage Declaration

AI was used for:

* [x] Understanding existing code
* [x] Generating code
* [x] Drafting documentation
* [x] Reviewing the diff
* [ ] Refactoring
* [ ] Generating tests

Human verification:

* [x] I understand every meaningful change in this PR
* [x] I checked generated code manually
* [x] I removed unsupported AI assumptions
* [x] I accept responsibility for this PR

---

## 8. Reviewer Focus

Please focus your review on:

* [x] Correctness
* [ ] Architecture
* [ ] Security
* [ ] Performance
* [ ] Tests
* [ ] Maintainability
* [ ] Product intent
* [ ] Edge cases

### CODE REVIEW

File: tools/dashboards/src/dashboards/envoy_ai_gateway/scoreboard.py [hunk 1]
```diff
@@ -286,18 +296,18 @@ def _hero() -> text.Panel:
 # ---------------------------------------------------------------------------
 
 
-def _stat_spend_30d() -> object:
+def _stat_spend_range() -> object:
     return sh.stat_panel(
-        title="Spend (last 30d)",
-        expr=sh.usd(f"sum(increase({METRIC_COST_MICRO_USD}{_SEL}[30d]))"),
+        title="Spend (range)",
+        expr=sh.usd(f"sum(increase({METRIC_COST_MICRO_USD}{_SEL}[$__range]))"),
```
[the remark this answers is not in the document]
Comment (stephane-segning) [reply tools/dashboards/src/dashboards/envoy_ai_gateway/scoreboard.py:302]: **False positive — meta-critique for @lightbridge-assistant's benefit.**

All 6 flagged locations (lines 159, 302, 341, 355, 367, 379) are calls of the
shape `sh.usd(f"...")` / `sh.stat_panel(...)` / `sh.bargauge_panel(...)`. The
`opengrep` rule `opt.opengrep-rules.python.sh.security.string-concat` appears
to pattern-match on the identifier `sh` and assume it refers to the `sh`
PyPI package (a shell-command wrapper, e.g. `sh.ls("-l")` shelling out) or a
`subprocess`-style shell invocation with f-string interpolation — the classic
injection shape the rule is designed for.

Here, `sh` is this file's local module alias:
`from dashboards.envoy_ai_gateway import _shared as sh` — this repo's
Python dependency set (`tools/dashboards/pyproject.toml`) is just
`grafana-foundation-sdk`; there is no `sh` package, no `subprocess`, no
shell invocation anywhere in this module. `sh.usd()` is a pure string
helper in `_shared.py`:

```python
def usd(expr: str) -> str:
    """Convert a micro-USD PromQL expression to USD (the cost counter is µ$)."""
    return f"(({expr}) / 1e6)"
```

It takes a PromQL expression string and wraps it in `(… / 1e6)` — no process
is ever spawned, no shell is ever invoked. The f-strings build **PromQL query
text** that gets embedded in a generated Grafana dashboard JSON, not a shell
command. `LABEL_DISPLAY_NAME`/`LABEL_MODEL`/`METRIC_COST_MICRO_USD`/`_SEL` are
all repo-internal constants (`dashboards/_common.py`), not user input, so
there's no injection surface even under the (wrong) shell-command reading.

Severity ("P1 security") is miscalibrated for the same reason across all 6
instances — this is a naming collision (our `sh` alias vs. the rule's assumed
`sh`-the-shell-library), not a security-relevant pattern. No change made;
verified by reading `_shared.py` and this repo's dependency list directly.

🤖 Verified and rebutted by Claude — human (@stephane-segning) reviewing.
